### PR TITLE
Serve blobs when a storage driver supports redirects but are disabled

### PR DESCRIPTION
**Pull request by BrianBland**
Fixes issue where an error was returned instead of serving the blob

#1300, docker/docker#15264

### CODE REVIEW

File: registry/storage/blobserver.go [hunk 1]
```diff
@@ -34,45 +34,45 @@ func (bs *blobServer) ServeBlob(ctx context.Context, w http.ResponseWriter, r *h
...
 			http.Redirect(w, r, redirectURL, http.StatusTemporaryRedirect)
 			return err
-		}
 
-	case driver.ErrUnsupportedMethod:
-		// Fallback to serving the content directly.
-		br, err := newFileReader(ctx, bs.driver, path, desc.Size)
-		if err != nil {
+		case driver.ErrUnsupportedMethod:
+			// Fallback to serving the content directly.
```
Comment (andrewnguyen) [inline registry/storage/blobserver.go]: This comment seems to be out of date now

Comment (BrianBland) [inline registry/storage/blobserver.go]: In the case of `driver.ErrUnsupportedMethod` we fall out of the type switch and then serve the content directly, so I think the comment is still valuable here.

**Comment by stevvooe** [comment]
@BrianBland This has nothing to do with redirect. This functionality should be handled by the `ServeBlob` method.

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**